firmware/applications/SignBuddy/flex.c

```diff
@@ -50,14 +51,15 @@ static inline void sample_data(void)
 
 static void flex_task(void *arg)
 {
+  LOG_INFO("flex: task started\r\n");
   while (1) {
     ulTaskNotifyTake(pdFALSE, portMAX_DELAY);
     sample_data();
     s.callback();
   }
 }
 
-void dma_data_ready_cb(void)
+void adc_data_ready_cb(void)
 {
   vTaskNotifyGiveFromISR(s.task_handle, 0);
```

Comment:
i think you should do this

      vTaskNotifyGiveFromISR(s.task_handle, &xHigherPriorityTaskWoken);
      portYIELD_FROM_ISR(xHigherPriorityTaskWoken);

Reply:
Sounds good, I'll address on that on a different branch